### ISSUE TITLE
Multiple issues

### ISSUE THREAD

**Issue by GoneD** [ghostserverd/mediaserver-docker#20]
Hi,

Thanks for your mediaserver-docker, but i have multiple issue : 

- When i configure Sonarr for use download client transmission and test to download torrent - is working well but when download is finish...Sonarr dont see the file (Filebot do nothing ???)
- If i try the cmd line from :7676 filebot i have this error : 
`Invalid usage: output folder [/share/media] must not be inside input folder [/share] and vice versa
Abort (×_×)
`
- Another issue portainer password not working - switch to portainer-ce to working well with reset password

Thanks for help ;)

**Comment by ghostserverd** [comment]
What does your directories configuration look like? This part in the .env file

```
# ======== directories ========
CONFIG_DIR=/some/directory/media_server/config
DOWNLOAD_DIR=/some/directory/media_server/downloads
MEDIA_DIR=/some/directory/media_server/media
TV_DIR=/some/directory/media_server/media/TV Shows
MOVIES_DIR=/some/directory/media_server/media/Movies
### base dir should be a directory that contains both downloads and media to take advantage of hardlinks
BASE_DIR=/some/directory/media_server
```

This message is telling us that the output folder must not be inside the input folder:

https://www.filebot.net/forums/viewtopic.php?t=9528

https://www.filebot.net/forums/viewtopic.php?t=3738

Make sure your `MEDIA_DIR` is not inside of your `DOWNLOAD_DIR`

**Comment by ghostserverd** [comment]
Can you open a separate issue for the portainer password? And include more details. How do you reproduce it? Password is working fine for me with the portainer container, but I also have had it set up for years now so maybe something is broken on initial setup.

**Comment by GoneD** [comment]
> What does your directories configuration look like? This part in the .env file
> 
> ```
> # ======== directories ========
> CONFIG_DIR=/some/directory/media_server/config
> DOWNLOAD_DIR=/some/directory/media_server/downloads
> MEDIA_DIR=/some/directory/media_server/media
> TV_DIR=/some/directory/media_server/media/TV Shows
> MOVIES_DIR=/some/directory/media_server/media/Movies
> ### base dir should be a directory that contains both downloads and media to take advantage of hardlinks
> BASE_DIR=/some/directory/media_server
> ```
> 
> This message is telling us that the output folder must not be inside the input folder:
> 
> https://www.filebot.net/forums/viewtopic.php?t=9528
> 
> https://www.filebot.net/forums/viewtopic.php?t=3738
> 
> Make sure your `MEDIA_DIR` is not inside of your `DOWNLOAD_DIR`

```
CONFIG_DIR=/data/seedbox/media_server/config
DOWNLOAD_DIR=/data/seedbox/media_server/downloads
MEDIA_DIR=/data/seedbox/media_server/media
TV_DIR=/data/seedbox/media_server/media/TV Shows
MOVIES_DIR=/data/seedbox/media_server/media/Movies
### base dir should be a directory that contains both downloads and media to take advantage of hardlinks
BASE_DIR=/data/seedbox/media_server
```

This is my .env settings.

For portainer - you can try your actual docker-compose to another host for example.

But i have read is replace by portainer/portainer-ce - your using portainer/portainer

Thanks,
Regards,

**Comment by ghostserverd** [comment]
In `/data/seedbox/media_server/config/transmission/` there should be a file called `filebot.log`. Can you attach it?

Also in `/data/seedbox/media_server/config/filebot/` there should be a file called `amc.log`. Could you attach that as well?

**Comment by GoneD** [comment]
> /data/seedbox/media_server/config/transmission

**chios :: media_server/config/filebot % cat amc.log**
```
Run script [fn:amc] at [Thu Dec 31 09:18:16 EST 2020]
Parameter: unsorted = n
Parameter: clean = y
Parameter: music = y
Parameter: artwork = y
Parameter: subtitles = en
Parameter: minLengthMS = 0
Parameter: movieFormat = {plex} - {hd}
Parameter: seriesFormat = {plex} - {hd}
Parameter: animeFormat = {plex} - {hd}
Parameter: musicFormat = {plex} - {hd}
Parameter: seriesDB = TheMovieDB::TV
Parameter: animeDB = AniDB
Parameter: movieDB = TheMovieDB
Parameter: musicDB = ID3
Parameter: ut_dir = /share
Parameter: ut_kind = multi
Parameter: ut_title =
Parameter: ut_label =
Invalid usage: output folder [/share/media] must not be inside input folder [/share] and vice versa
Abort (×_×)
```
I dont found filebot.log... is not exist :(
```
chios :: seedbox/media_server/config % locate filebot.log
chios :: seedbox/media_server/config %                           

```

**Comment by GoneD** [comment]
Hi,
Do you have any idea or other thing about my issue ?
Thanks :)

**Comment by ghostserverd** [comment]
This is the problem

> Parameter: ut_dir = /share

we need to figure out why the post process script is setting ur_dir to /share instead of the directory of the completed download.

**Comment by nielsvanhooy** [comment]
last comment got me in the right direction why it wasnt working. i changed the filebot command manually and put in a bash file ->  put a cronjob on the host system to just run every 4 minutes. wife happy now at least.

docker exec -it filebot filebot -script fn:amc --output /share/media --action move --conflict auto --filter 'any{ age < 30 }{ true } || !model.any{ it.age < 30 && it.age != null }' -non-strict -no-xattr --log-file /config/amc.log --def unsorted=n clean=y music=y artwork=y subtitles=en minLengthMS=0 movieFormat="{plex} - {hd}" seriesFormat="{plex} - {hd}" animeFormat="{plex} - {hd}" musicFormat="{plex} - {hd}" seriesDB="TheMovieDB::TV" animeDB="AniDB" movieDB="TheMovieDB" musicDB="ID3" ut_dir="$(eval echo '/downloads/complete/sonarr/$v_path')" ut_kind="multi" ut_title="$(eval echo '$v_name')" ut_label="$(eval echo '$v_label')"

i just changd the ut_dir info (left the env) to my direct completed download folder.

**Comment by GoneD** [comment]
Thanks for this - but how just to modifiy the container to using ut_dir correctly ?

**Comment by GoneD** [comment]
Ok i have modify the script inside the filebot container : 

```
root@xxxxxx:/etc/services.d/filebot# ls
run
```

nano **run** and add this :

`UTDIR_NEW_PATH=${UTDIR_NEW_PATH:-/downloads/complete/sonarr}
`

And modify this : 

```
# necessary to take advantage of hardlinks
DIR_FULL_PATH="$UTDIR_NEW_PATH$UT_DIR"
```

:)

**Comment by GoneD** [comment]
@ghostserverd  - how sonarr or radarr call filebot ? because never append - if i launch manualy is ok - but is not automaticly launch after download ? or is transmision ?

**Comment by carefreecaribou** [comment]
@ghostserverd

I seem to be having this same issue too. Same type of directory setup.
The start of my .env:

```
# ======== directories ========
CONFIG_DIR=/home/toast/mediaserver-docker/configs
DOWNLOAD_DIR=/plexmedia/drive1/downloads
MEDIA_DIR=/plexmedia/drive1/media
TV_DIR=/plexmedia/drive1/media/TV
MOVIES_DIR=/plexmedia/drive1/media/Movies
### base dir should be a directory that contains both downloads and media to take advantage of hardlinks
BASE_DIR=/plexmedia/drive1
```

My filebot.log:

```
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/radarr/[REDACTED NAME]" --data-urlencode label="movie" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc

```

The last entry (a movie collection) worked just fine. It was added manually to transmission not through Sonarr or Radarr and was downloaded to /plexmedia/drive1/downloads/complete rather than /plexmedia/drive1/downloads/complete/sonarr for downloads from sonarr. It appeared to stop working for me after I added the Sonarr tag to sonarr downloads. I tried manually moving the downloads to the same directory of that movie, but filebot still errored out. My most recent AMC error:

```
Run script [fn:amc] at [Fri Jan 15 14:27:15 UTC 2021]
Parameter: unsorted = n
Parameter: clean = y
Parameter: music = y
Parameter: artwork = y
Parameter: subtitles = en
Parameter: minLengthMS = 0
Parameter: movieFormat = {plex} - {hd}
Parameter: seriesFormat = {plex} - {hd}
Parameter: animeFormat = {plex} - {hd}
Parameter: musicFormat = {plex} - {hd}
Parameter: seriesDB = TheMovieDB::TV
Parameter: animeDB = AniDB
Parameter: movieDB = TheMovieDB
Parameter: musicDB = ID3
Parameter: ut_dir = /share
Parameter: ut_kind = multi
Parameter: ut_title = 
Parameter: ut_label = 
Invalid usage: output folder [/share/media] must not be inside input folder [/share] and vice versa
Abort (×_×)
```

**Comment by GoneD** [comment]
> @ghostserverd
> 
> I seem to be having this same issue too. Same type of directory setup.
> The start of my .env:
> 
> ```
> # ======== directories ========
> CONFIG_DIR=/home/toast/mediaserver-docker/configs
> DOWNLOAD_DIR=/plexmedia/drive1/downloads
> MEDIA_DIR=/plexmedia/drive1/media
> TV_DIR=/plexmedia/drive1/media/TV
> MOVIES_DIR=/plexmedia/drive1/media/Movies
> ### base dir should be a directory that contains both downloads and media to take advantage of hardlinks
> BASE_DIR=/plexmedia/drive1
> ```
> 
> My filebot.log:
> 
> ```
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/radarr/[REDACTED NAME]" --data-urlencode label="movie" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/sonarr/[REDACTED NAME]" --data-urlencode label="tv" http://filebot:[redactedport]/amc
> curl --data-urlencode name="[REDACTED NAME]" --data-urlencode path="/downloads/complete/[REDACTED NAME]" --data-urlencode label="N/A" http://filebot:[redactedport]/amc
> ```
> 
> The last entry (a movie collection) worked just fine. It was added manually to transmission not through Sonarr or Radarr and was downloaded to /plexmedia/drive1/downloads/complete rather than /plexmedia/drive1/downloads/complete/sonarr for downloads from sonarr. It appeared to stop working for me after I added the Sonarr tag to sonarr downloads. I tried manually moving the downloads to the same directory of that movie, but filebot still errored out. My most recent AMC error:
> 
> ```
> Run script [fn:amc] at [Fri Jan 15 14:27:15 UTC 2021]
> Parameter: unsorted = n
> Parameter: clean = y
> Parameter: music = y
> Parameter: artwork = y
> Parameter: subtitles = en
> Parameter: minLengthMS = 0
> Parameter: movieFormat = {plex} - {hd}
> Parameter: seriesFormat = {plex} - {hd}
> Parameter: animeFormat = {plex} - {hd}
> Parameter: musicFormat = {plex} - {hd}
> Parameter: seriesDB = TheMovieDB::TV
> Parameter: animeDB = AniDB
> Parameter: movieDB = TheMovieDB
> Parameter: musicDB = ID3
> Parameter: ut_dir = /share
> Parameter: ut_kind = multi
> Parameter: ut_title = 
> Parameter: ut_label = 
> Invalid usage: output folder [/share/media] must not be inside input folder [/share] and vice versa
> Abort (×_×)
> ```

Look at my answer above you have the solution.

**Comment by ghostserverd** [comment]
It's still not clear to me what's going on between the transmission postprocess script and filebot that's causing this.

The `BASE_DIR` is mounted to `/share` in the `filebot` container

The `DOWNLOAD_DIR` is mounted to `/downloads` in the `filebot` container

E.g.

`/plexmedia/drive1` is mounted to `/share`
`/plexmedia/drive1/downloads` is mounted to `/downloads`

`UT_DIR='$v_path'` will be whatever was passed into the command

so `UT_DIR` is `/downloads/complete/[REDACTED NAME]`

`SHARE_BASE_PATH` is `/share`

So the resulting values should be

```
DIR_FULL_PATH="$SHARE_BASE_PATH$UT_DIR"
MEDIA_FULL_PATH="$SHARE_BASE_PATH/media"
```

or once the replacement happens

```
DIR_FULL_PATH="/share/downloads/complete/sonarr/[REDACTED NAME]"
MEDIA_FULL_PATH="/share/media"
```

The fact that the `amc.log` is showing `/share` means that `$UT_DIR` is empty for some reason.

And it seems to only affect `sonarr` and `radarr` tags. So it must be something in the post processing script that's setting the `path` to be empty, but it's lying about what it set it to in the logs.

**Comment by mikeczabator** [comment]
> Ok i have modify the script inside the filebot container :
> 
> ```
> root@xxxxxx:/etc/services.d/filebot# ls
> run
> ```
> 
> nano **run** and add this :
> 
> `UTDIR_NEW_PATH=${UTDIR_NEW_PATH:-/downloads/complete/sonarr} `
> 
> And modify this :
> 
> ```
> # necessary to take advantage of hardlinks
> DIR_FULL_PATH="$UTDIR_NEW_PATH$UT_DIR"
> ```
> 
> :)

I made those changes, still no luck.  

:-(

**Comment by IncognitoCoding** [comment]
**Issue Explained:**
I hit the same issue using the "ghostserverd/transmission" docker image. The problem is the "transmission-postprocess.sh" post-processing script is not getting any parameters passed once the torrent finishes ($TR_TORRENT_DIR, $TR_TORRENT_NAME, etc). This in return causes issues with the filebot call further into the "transmission-postprocess.sh" script.

**Original Posted Fix Suggestion**
Editing the "run" file in the filebot container will allow the process to run, but it will not know which files to process and you will get errors if you have existing torrent files. I do not recommend this route. These missing parameters tell filebot which torrent to process, so it will skip over existing torrents.

**Resolution Option 1:**
If you want to continue using the "ghostserverd/transmission" docker image, you will have to wait for ghostserverd to update the docker image for transmission.

**Resolution Option 2:**
If you want to use a newer release that works you can use "ghcr.io/linuxserver/transmission" instead of "ghostserverd/transmission". Notice they do use a different port number for their image. Also, some additional options in the "ghostserverd" image may need to be added to the "settings.json" file if you want them pre-configured.

Setup notes:
> Step 1: Copy the transmission-postprocess.sh file or text from "ghostserverd/transmission" docker image (Path: /usr/local/bin/transmission-postprocess.sh) and add it into your bind mapped location. (ex: CONFIG_DIR=/some/directory/media_server/config/transmission/transmission-postprocess.sh)
> Step 2: On your docker host run "chmod +x transmission-postprocess.sh". You will get access denied during execution without completing this step.
> Step 3: If you have an existing mapped file (CONFIG_DIR=/some/directory/media_server/config/transmission/settings.json) delete or change the name (ex: .bak) of that file on your host before starting the container. 
> Step 4: Start your updated docker-compose file. Once the container starts it will create a new "settings.json" file. 
> Step 5: Edit your new settings.json file and update the "script-torrent-done-*" options.
> Step 5.1: "script-torrent-done-enabled": true,
> Step 5.2: "script-torrent-done-filename": "/config/transmission-postprocess.sh",
> Step 6 (Optional): If you are using a different VPN container other than "ghostserverd/wireguard" and you have issues with internal DNS resolutions for transmission you can update the URL on transmission-postprocess.sh script directly.

**Comment by ghostserverd** [comment]
`ghostserverd/transmission` builds `FROM linuxserver/transmission` and the docker hub configuration is set to auto-build when there is a change to `linuxserver/transmission`, so it's up to date with `linuxserver/transmission`.

I'm still unable to reproduce this behavior. @IncognitoCoding do you have a `filebot.log` and `amc.log` that you could provide since it sounds like you're able to reproduce this?

For reference here's a recent `filebot.log` from my own setup

```
curl --data-urlencode name="Kims.Convenience.S05E04.1080p" --data-urlencode path="/downloads/complete/sonarr/Kims.Convenience.S05E04.1080p" --data-urlencode label="tv" http://filebot:7676/amc
```

and the corresponding entries in `amc.log`

```
Run script [fn:amc] at [Sun Feb 14 01:33:24 UTC 2021]
Parameter: unsorted = n
Parameter: clean = y
Parameter: music = y
Parameter: artwork = y
Parameter: subtitles = en
Parameter: minLengthMS = 0
Parameter: movieFormat = {plex} - {hd}
Parameter: seriesFormat = {plex} - {hd}
Parameter: animeFormat = {plex} - {hd}
Parameter: musicFormat = {plex} - {hd}
Parameter: seriesDB = TheTVDB
Parameter: animeDB = AniDB
Parameter: movieDB = TheMovieDB
Parameter: musicDB = ID3
Parameter: ut_dir = /share/downloads/complete/sonarr/Kims.Convenience.S05E04.1080p
Parameter: ut_kind = multi
Parameter: ut_title = Kims.Convenience.S05E04.1080p
Parameter: ut_label = tv
Input: /share/downloads/complete/sonarr/Kims.Convenience.S05E04.1080p/Kims.Convenience.S05E04.1080p.mkv
Process as TV Series [tv]
Group: {Series=true} => [Kims.Convenience.S05E04.1080p.mkv]
Get [English] subtitles for 1 files
CmdlineException: OpenSubtitles: Please enter your login details by calling `filebot -script fn:configure`
Rename episodes using [TheTVDB] with [Airdate Order]
Auto-detected query: [Kims Convenience]
Fetching episode data for [Kim's Convenience]
Apply filter [any{ age < 30 }{ true } || !model.any{ it.age < 30 && it.age != null } || true] on [59] options
[59 / 59] options passed
[DUPLICATE] from [/share/downloads/complete/sonarr/Kims.Convenience.S05E04.1080p/Kims.Convenience.S05E04.1080p.mkv] to [/share/media/TV Shows/Kim's Convenience/Season 05/Kim's Convenience - S05E04 - Tennis Anyone - HD.mkv]
Processed 1 files
Fetching series artwork for [Kim's Convenience / Season 5] to [/share/media/TV Shows/Kim's Convenience/Season 05]
Generate Series NFO: Kim's Convenience [315908]
Banner already exists: /share/media/TV Shows/Kim's Convenience/poster.jpg
Banner already exists: /share/media/TV Shows/Kim's Convenience/banner.jpg
Banner already exists: /share/media/TV Shows/Kim's Convenience/fanart.jpg
Banner not found: /share/media/TV Shows/Kim's Convenience/Season 05/poster.jpg / season:season
Banner not found: /share/media/TV Shows/Kim's Convenience/Season 05/banner.jpg / seasonwide:seasonwide
Fanart not found: /share/media/TV Shows/Kim's Convenience/clearart.png / hdclearart
Fanart not found: /share/media/TV Shows/Kim's Convenience/clearart.png / clearart
Fanart not found: /share/media/TV Shows/Kim's Convenience/logo.png / hdtvlogo
Fanart not found: /share/media/TV Shows/Kim's Convenience/logo.png / clearlogo
Fanart not found: /share/media/TV Shows/Kim's Convenience/landscape.jpg / tvthumb
Fanart not found: /share/media/TV Shows/Kim's Convenience/Season 05/landscape.jpg / seasonthumb
Done ヾ(＠⌒ー⌒＠)ノ
```

**Comment by IncognitoCoding** [comment]
**@ghostserverd** I was able to reproduce it the day I originally posted this post, but I do not have the original logs and I have shifted over to using the linuxserver.io transmission release. I like having manual control over adjusting the transmission-postprocess.sh script. It allows ease of adding the IP of my docker to FileBot, Sonarr, and Radarr instead of doing the other options to get the DNS to resolve when routing through Wireguard. My only guess is something on your customized image is causing an issue passing the transmission complete parameters ($TR_TORRENT_DIR, $TR_TORRENT_NAME, etc) over when calling the "transmission-postprocess.sh" script for certain setups.

All I can provide is a little around my original setup. I wasn't using your environment file and had some different folder path structures, but the problem was still the same result as a previous poster that was using your .env file. I was running everything on a single VM running Ubuntu 20.04 with the latest updates.

I am using your Filebot image and it works great! I get consistent results and no issues.

My brief testing wasn't allowing FileBot to work or port forwarding when adding the port into your Transmission. The linuxserver.io version worked on both, so I wrote a custom script to automate updating the port with a cronjob and called it good. If I didn't have to redo everything I would gladly try and recreate the issue for you. Hopefully, another poster will be able to provide you some additional logging if they hit the same issue.

**Comment by ghostserverd** [comment]
@IncognitoCoding understood. Glad you were able to get it set up with the linuxserver image anyway.

@thetoastedghost if you want to keep trying to troubleshoot this, could you provide the entire `filebot.log`? None of the log entries that you posted show an empty `path` or `name` in the request being made to `filebot`, so I don't think any of those line up with the issue you're seeing.

I added some additional logging to print the variables in the post process script as well. So if you run into this again, could you also include the `transmission_env.log` file on this issue? That should hopefully help with troubleshooting.

**Comment by nhisoto** [comment]
Hi, Thanks for the repo. It is mostly working for me. I ran into the same issue as @GoneD and was able to fix it by editing the "run" file withing the filebot container.
I tried @IncognitoCoding solution too but that did not work for me.
Have you ran into this at all or been able to replicate the issue?

**Comment by ghostserverd** [comment]
> Hi, Thanks for the repo. It is mostly working for me. I ran into the same issue as @GoneD and was able to fix it by editing the "run" file withing the filebot container.
> I tried @IncognitoCoding solution too but that did not work for me.
> Have you ran into this at all or been able to replicate the issue?


I have not had any luck reproducing this. You said you were `able to fix it by editing the "run" file withing the filebot container.`. What changes did you make specifically?

If you have a repro, could you provide your `filebot.log`, `transmission.log`, and `transmission_env.log` and I'll take a look to see if anything stands out?

Thanks!

**Comment by aspopovic** [comment]
I still have the same issue today. Any news on solution, as GoneD's works only for sonarr, not for radarr, and IncognitoCodings is just too complicated... This is one of the most complete solutions and it kills me not being able to do it... I didn't try the cronjob though, since it again shoots on sonarr only...

**Comment by ghostserverd** [comment]
It’s very much still “works on my machine” so any troubleshooting details you can provide would be helpful.

Could you upload your filebot.log, transmission.log, and transmission_env.log as well as your .env file?

**Comment by aspopovic** [comment]
> It’s very much still “works on my machine” so any troubleshooting details you can provide would be helpful.
> 
> Could you upload your filebot.log, transmission.log, and transmission_env.log as well as your .env file?

Okay, now I'm puzzled. I've done a number of things last night trying to "fix" filebot. I've changed my partition to ext4 (took 12 hours, lol), and god knows what else: changed directory names to lowercase, put configs directory in the same place as downloads and media, and moved them all from disk root to media_server directory. After work, I've tried few radarr downloads and filebot seems to work fine now, damn... I'm trying sonarr download right now and if it works I'll consider it fixed. When i run amc from shell2http manually, it still fails though, probably because it doesn't have an idea what's where. Thank you for all your hard work 👍

**Comment by patcfly** [comment]
I believe I am hitting this same issue. Transmission downloads work fine, anything from nzbget + sonarr or + radarr does not work properly #36 is effectively a duplicate of this isssue.

I have even tried doing a net-new clone of the ghostserverd repo and built my environment up from scratch with a new base directory ( /mnt/md0/media_server vs /mnt/md0/mediaserver).

I didn't even make any of the child dirs under that base directory - I made sure to set the base directory with necessary permissions so that the user ID 1000 owned and had full permissions on it. All directories are then made when I first run docker compose up.

Based on wat @calinorg said.. Maybe my issue is that I am doing this from a dir under /mnt/ on the hose instead of mounting the raid volume directly somewhere else in my hosts's file structure. I previously had ghostserverd up and running fine w/ sonarr and radarr when I had a raid volume mounted under /downloads or /data (cant remeber exactly which).

What is really interesting for me is that movies show up in plex even though radarr neverr shows them as complete. Sonarr initiated series downloads never import or show up in plex
[_nzbget_logs.txt](https://github.com/ghostserverd/mediaserver-docker/files/9343896/_nzbget_logs.txt)
[_sonarr_logs.txt](https://github.com/ghostserverd/mediaserver-docker/files/9343897/_sonarr_logs.txt)
[_filebot_logs.txt](https://github.com/ghostserverd/mediaserver-docker/files/9343898/_filebot_logs.txt)
.

**Comment by patcfly** [comment]
moved my base dir to be /data/media_server

still having the issue
[_filebot_logs (1).txt](https://github.com/ghostserverd/mediaserver-docker/files/9343945/_filebot_logs.1.txt)

**Comment by amrollz** [comment]
Hate to necro a thread, but I too am experiencing the issues listed above.

When I attempt to run /usr/local/bin/transmission-postprocess.sh manually I get the following error "/usr/local/bin/transmission-postprocess.sh: line 4: TR_TORRENT_DIR: parameter not set".

It was working fine until 3 or 4 days ago, but since that point filebot seemed to cease to function.
